### PR TITLE
Starting a CLI for relevancy.

### PR DESCRIPTION
**Pull request by bendk**
This can be run with `cargo relevancy`.

It currently supports classifying your history and printing the results. The main way we get history is from sync.  You can also point it to a places DB, but this must be from a mobile FF, we can't read from desktop.

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due diligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

File: examples/relevancy-cli/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "examples-relevancy-cli"
+version = "0.1.0"
+authors = ["sync-team@mozilla.com"]
```
Comment (ncloudioj) [inline examples/relevancy-cli/Cargo.toml:4]: Homesick? ;)

File: examples/relevancy-cli/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+#[derive(Parser)]
+#[command(about, long_about = None)]
+struct Cli {
+    /// Printout extra details, including how each URL is classified
+    #[clap(long, short, action)]
+    verbose: bool,
+
+    /// Load places data from disk rather than syncing it
+    ///
+    /// Note: this only works mobile places databases, desktop uses a different format.
```
Comment (ncloudioj) [inline examples/relevancy-cli/src/main.rs:29]:
```suggestion
    /// Note: this only works for mobile places databases, desktop uses a different format.
```

**Review by ncloudioj — APPROVED** [review]
Not sure about the sync magic, but it looks cool.

File: examples/relevancy-cli/Cargo.toml (hunk 1 above)
Comment (bendk) [reply examples/relevancy-cli/Cargo.toml:4]: Oops, copy and paste error.  I think this is an optional field so I'm just going to remove it.

**Pull request merged**